pkg/query-service/app/querier/querier.go

```diff
@@ -160,11 +160,15 @@ func findMissingTimeRanges(start, end int64, seriesList []*v3.Series, fluxInterv
 		}
 	}
 
+	endMillis := time.Now().UnixMilli()
```

Comment:
can this be extracted into a function as I see multiple occurrences of this adjustment logic? If we need to change this again later, we should need not look for different places or miss it at some places in hurry

Reply:
I can do that